Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2680,13 +2683,16 @@ extension BrowserViewController: FeatureFlaggable {
 }
 
 extension BrowserViewController {
+    /// This method now returns the BrowserViewController associated with the scene.
+    /// We currently have a single scene app setup, so this will change as we introduce support for multiple scenes.
     public static func foregroundBVC() -> BrowserViewController {
-        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate,
-              let browserViewController = appDelegate.browserViewController else {
-            fatalError("Unable unwrap BrowserViewController")
+        guard let sceneDelegate = UIApplication.shared.connectedScenes.first?.delegate as? SceneDelegate else {
+            /// Currently, we have a single scene app. If we're here, there's no scene or window.
+            /// This should be impossible, and fatal.
+            fatalError("Unable to fetch the Scene.")
         }
 
-        return browserViewController
+        return sceneDelegate.browserViewController
     }
```

Comment:
Thinking about this a bit more, whenever we foregroundBVC, I'd expect to get the BVC of the actively interacted with scene. But we'll know for sure when we start supporting multiple windows.